### ISSUE TITLE
diesel-cli 1.4.4 doesn't build with rust 1.42.0

### ISSUE THREAD

**Issue by kpcyrd** [diesel-rs/diesel#2337]
Building the most recent version of diesel-cli for archlinux fails since we do not build with rustup and the `rust-toolchain` file is therefore ignored.

`#[deny(warnings)]` is not future proof and should be removed. It should be sufficient to enforce this during CI builds.

```
   Compiling diesel_derives v1.4.1 (/build/diesel-cli/src/diesel-1.4.4/diesel_derives)
   Compiling serde_derive v1.0.105
   Compiling dotenv v0.10.1
   Compiling diesel v1.4.4 (/build/diesel-cli/src/diesel-1.4.4/diesel)
   Compiling toml v0.4.10
error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/result.rs:269:64
    |
269 |             Error::RollbackTransaction => write!(f, "{}", self.description()),
    |                                                                ^^^^^^^^^^^
    |
note: lint level defined here
   --> diesel/src/lib.rs:134:5
    |
134 |     warnings,
    |     ^^^^^^^^
    = note: `#[deny(deprecated)]` implied by `#[deny(warnings)]`

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/result.rs:270:65
    |
270 |             Error::AlreadyInTransaction => write!(f, "{}", self.description()),
    |                                                                 ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/result.rs:279:59
    |
279 |             Error::InvalidCString(ref nul_err) => nul_err.description(),
    |                                                           ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/result.rs:282:50
    |
282 |             Error::QueryBuilderError(ref e) => e.description(),
    |                                                  ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/result.rs:283:53
    |
283 |             Error::DeserializationError(ref e) => e.description(),
    |                                                     ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/result.rs:284:51
    |
284 |             Error::SerializationError(ref e) => e.description(),
    |                                                   ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/result.rs:319:69
    |
319 |             ConnectionError::InvalidCString(ref nul_err) => nul_err.description(),
    |                                                                     ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/result.rs:322:68
    |
322 |             ConnectionError::CouldntSetupConfiguration(ref e) => e.description(),
    |                                                                    ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/result.rs:368:30
    |
368 |         write!(f, "{}", self.description())
    |                              ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
  --> diesel/src/migration/errors.rs:40:57
   |
40 |             MigrationError::IoError(ref error) => error.description(),
   |                                                         ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
  --> diesel/src/migration/errors.rs:54:14
   |
54 |         self.description().fmt(f)
   |              ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
  --> diesel/src/migration/errors.rs:98:68
   |
98 |             RunMigrationsError::MigrationError(ref error) => error.description(),
   |                                                                    ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
  --> diesel/src/migration/errors.rs:99:64
   |
99 |             RunMigrationsError::QueryError(ref error) => error.description(),
   |                                                                ^^^^^^^^^^^

error: use of deprecated item 'std::error::Error::description': use the Display impl or to_string()
   --> diesel/src/migration/errors.rs:108:43
    |
108 |         write!(f, "Failed with: {}", self.description())
    |                                           ^^^^^^^^^^^

error: aborting due to 14 previous errors

error: could not compile `diesel`.
warning: build failed, waiting for other jobs to finish...
error: build failed
==> ERROR: A failure occurred in build().
    Aborting...
==> ERROR: Build failed, check /var/lib/archbuild/extra-x86_64/kpcyrd/build
```

**Comment by weiznich** [comment]
Please provide more information about how to reproduce that. `cargo +1.42.0 install diesel_cli` works fine for me. If you are trying to build from local source there is a rust-toolchain file that says what rustc version is expected to be used by default. Otherwise you can also disable lints by using `RUSTFLAGS` (what cargo does internally).

**Comment by kpcyrd** [comment]
We're building diesel-cli with `cargo build --release` from the release tarball and /usr/bin/rustc is the actual compiler instead of a symlink to rustup.

```
RUSTFLAGS="-A warnings" cargo build --release
```
doesn't work, it seems `#[deny(warnings)]` takes precedence. Why not set `RUSTFLAGS="-D warnings"` in the CI instead?

**Comment by weiznich** [comment]
> Why not set RUSTFLAGS="-D warnings" in the CI instead?

Because then the CI would also fail if a third party dependency generates a warning. `RUSTFLAGS` applies to all crates that a compiled at a certain point, not only the root crate. Using `#[deny(warning)]` gives us a finer control there.

>We're building diesel-cli with cargo build --release from the release tarball and /usr/bin/rustc is the actual compiler instead of a symlink to rustup.
> ```
> RUSTFLAGS="-A warnings" cargo build --release
> ```

Have you tried using `--cap-lints allow` as `RUSTFLAGS`, because that's what cargo is doing internally and that is working.

That said I'm not opposed to getting a fix for the cause of this problem as long as it does not result in any degradation of the development process (issue errors for warnings in dependencies would be a problem).

**Comment by weiznich** [comment]
Fixed with the release of diesel_cli 1.4.1
